frb_codegen/src/main.rs

```diff
@@ -14,20 +12,21 @@ fn main() -> anyhow::Result<()> {
     #[cfg(feature = "serde")]
     let dump_config = raw_opts.dump.clone();
 
-    let configs = config_parse(raw_opts);
-    debug!("configs={:?}", configs);
+    let (all_configs, all_symbols) = config_parse(raw_opts)?;
+    debug!("configs={:?}", all_configs);
 
+    // generation of rust api for ffi
     #[cfg(feature = "serde")]
     if let Some(dump) = dump_config {
-        return dump::dump_multi(&configs, dump).context("Failed to dump config");
+        return dump::dump_multi(&all_configs, dump).context("Failed to dump config");
     }
 
-    // generation of rust api for ffi
-    let all_symbols = get_symbols_if_no_duplicates(&configs)?;
-
+    // If in multi-blocks case, the shared block MUST be generated at first.
+    // Otherwise, the generated dart code for regular blocks would be problematic, since when
+    // the shared rust module has not yet been generated.
     let mut errors = vec![];
-    for (config_index, config) in configs.iter().enumerate() {
-        if let Err(err) = frb_codegen_multi(&configs, config_index, &all_symbols) {
+    for (config_index, config) in (all_configs.iter().enumerate()).rev() {
```

Comment:
nit: I am not sure whether `rev` is the best way to ensure shared block is generated first, since this implicitly assume `shared` block is the last entry of all_configs, which can be easily broken.

Simple brainstorming:

```
struct Configs {
  shared_block_config: Config,
  normal_block_configs: Vec<Config>,
}

impl Configs {
  // ... comments telling readers that we really need "shared block" be first
  fn all() -> Vec<Config> {
    [self.shared_block_config, ...self.normal_block_configs]
  }
}
```